### PR TITLE
Duet AI ML data processing prompts (no links)

### PR DESCRIPTION
**Pull request by dariabezkorovaina**
Adding 3 prompts for ML data processing

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://github.com/apache/beam/blob/master/CONTRIBUTING.md#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI or the [workflows README](https://github.com/apache/beam/blob/master/.github/workflows/README.md) to see a list of phrases to trigger workflows.

### CODE REVIEW

**Comment by olehborysevych** [comment]
@dariabezkorovaina could you please open this PR?

File: learning/prompts/documentation-lookup-nolinks/34_ml_data_preprocessing.md [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+In this example, the Apache Beam pipeline performs the following steps:
+* Creates data.
+* Cleans data by filtering missing values.
+* Transforms data by scaling it.
+* Enriches data by adding coordinates from an external source.
+* Collects metrics and counts data instances.
+* Writes the processed data to an output file.
+
+In addition to standard data processing transforms, Apache Beam also provides a set of specialized transforms for preprocessing and transforming data, consolidating them into the `MLTransform` class. This class simplifies your workflow and ensures data consistency by enabling the use of the same steps for training and inference. You can use `MLTransform` to generate text embeddings and implement specialized processing modules provided by the TensorFlow Transforms (TFT) library for machine learning tasks.
```
Comment (damccorm) [inline learning/prompts/documentation-lookup-nolinks/34_ml_data_preprocessing.md:85]:
```suggestion
In addition to standard data processing transforms, Apache Beam also provides a set of specialized transforms for preprocessing and transforming data, consolidating them into the `MLTransform` class. This class simplifies your workflow and ensures data consistency by enabling the use of the same steps for training and inference. You can use `MLTransform` to generate text embeddings and implement specialized processing modules provided by the TensorFlow Transforms (TFT) library for machine learning tasks like computing and applying a vocabulary, scaling your data using z-scores, bucketizing your data, and more.
```

**Review by damccorm — COMMENTED** [review]
Thanks, just had one suggestion

**Comment by dariabezkorovaina** [comment]
@damccorm thank you, implemented! :)

**Pull request merged**